addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -2076,6 +2076,7 @@ MockServer.include({
                 "id": partner.id,
                 "im_status": partner.im_status,
                 "name": partner.name,
+                "user_id": partner.user_id,
```

Comment:
`user_id` field on partner model in python means something else unfortunately, so we can't do it this way.

Mock server should be a copy of what is done in python, in particular here we should look at how `user_id` is computed in `mail_partner_format`. We can make simplifications here though (for example ignore the check on internal user for now).